pkg/cloud/services/s3/s3.go

```diff
@@ -173,27 +173,35 @@ func (s *Service) Delete(m *scope.MachineScope) error {
 	bucket := s.bucketName()
 	key := s.bootstrapDataKey(m)
 
-	s.scope.Info("Deleting object", "bucket_name", bucket, "key", key)
-
-	_, err := s.S3Client.DeleteObject(&s3.DeleteObjectInput{
+	_, err := s.S3Client.HeadObject(&s3.HeadObjectInput{
```

Comment:
`HeadObject` must be added as allowed operation, or else the controller will fail on upgrade since users don't have this allowed in their IAM policy (see `cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go` – and maybe other places??).

This should also go into release notes (your PR has no release note at the moment).

Reply:
S3 is weird in that not all operations require a permission of the same name. In the case of HeadObject, it only requires GetObject permission: https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html

Reply:
I didn't realize the controller didn't already have this permission. Added.